onboarding/source/amb_light_service.c

```diff
@@ -18,22 +28,86 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    if(lightServiceTaskHandle == NULL){
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                &lightServiceTaskHandle);
 
+    }
+
+    if(xLightServiceQueue == NULL){
+        xLightServiceQueue = xQueueCreate(QUEUE_SiZE, LIGHT_EVENT_SIZE);
+    }
     /* USER CODE END */
     return 1;
 }
 
+/**
+ * @brief Get light data from adcData
+ * @param void
+ **/
+uint8_t* Get_Light_Sensor_data(void)
+{
+
+	adcData_t *adc_data_ptr = pvPortMalloc(sizeof(adcData_t));
+    adc_data_ptr->id = 6;
+    //adc_data_ptr->value;
+
+ 	/** - Start Group1 ADC Conversion 
+ 	*     Select Channel 6 - Light Sensor for Conversion
+ 	*/
+	adcStartConversion(adcREG1, adcGROUP1);
+
+ 	/** - Wait for ADC Group1 conversion to complete */
+ 	while(!adcIsConversionComplete(adcREG1, adcGROUP1)); 
+
+	/** - Read the conversion result
+	*     The data contains the Light sensor data
+    */
+	adcGetData(adcREG1, adcGROUP1, adc_data_ptr);
+	
+	/** - Transmit the Conversion data to PC using SCI
+    */
+	return (adc_data_ptr->value);
+}
+
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
-
+    ASSERT(lightServiceTaskHandle != NULL);
+    uint8_t lightServiceStatus = initLightService();
+    
+    if(lightServiceStatus == 0){
+        
+        printf("Failed to create a queue");
+        
+    }else{
+        
```

Comment:
Need to wait for light event in queue with xQueueReceive, also everything inside the else statement should be in an infinite loop.